--- denemo-0.8.6/src/utils.c~	2009-07-01 16:17:15.000000000 +0200
+++ denemo-0.8.6/src/utils.c	2009-07-07 14:44:35.176798844 +0200
@@ -678,7 +678,7 @@ get_locale_dir ()
   {
 #ifdef G_OS_WIN32
     gchar *rootdir = g_win32_get_package_installation_directory (NULL, NULL);
-    localedir = g_build_filename (rootdir, "share", "locale", "denemo", NULL);
+    localedir = g_build_filename (rootdir, "share", "locale", NULL);
     g_free (rootdir);
 #else /* not G_OS_WIN32 */
 # ifndef ENABLE_BINRELOC
