--- src/gp_unix.c~	2005-03-08 21:46:07.000000000 +0100
+++ src/gp_unix.c	2005-12-23 18:40:07.000000000 +0100
@@ -53,6 +53,7 @@
 #include <dlfcn.h>
 #include <string.h>
 
+
 void
 gp_init(void)
 {
@@ -135,6 +136,9 @@
 {
     struct timeval tp;
 
+    tp.tv_sec = tp.tv_usec = 0;
+    
+#ifdef HAVE_GETTIMEOFDAY
 #if gettimeofday_no_timezone	/* older versions of SVR4 */
     {
 	if (gettimeofday(&tp) == -1) {
@@ -150,7 +154,8 @@
 	}
     }
 #endif
-
+#endif
+    
     /* tp.tv_sec is #secs since Jan 1, 1970 */
     pdt[0] = tp.tv_sec;
 
--- src/gdevsgi.c~	2005-01-19 05:08:41.000000000 +0100
+++ src/gdevsgi.c	2005-12-23 18:39:20.000000000 +0100
@@ -76,7 +76,8 @@
 
      if ((data == (byte*)0)||(header == (IMAGE*)0)) return -1;
 
-     bzero(header,sizeof(IMAGE));
+     //     bzero(header,sizeof(IMAGE));
+     memset(header,0,sizeof(IMAGE));     
      header->imagic = IMAGIC;
      header->type = RLE(1);
      header->dim = 3;
--- src/gdevpipe.c~	2005-01-19 05:08:41.000000000 +0100
+++ src/gdevpipe.c	2005-12-23 18:38:17.000000000 +0100
@@ -16,6 +16,8 @@
 
 /* $Id: gdevpipe.c,v 1.6 2002/02/21 22:24:51 giles Exp $ */
 /* %pipe% IODevice */
+#include <stdlib.h>
+
 #include "errno_.h"
 #include "pipe_.h"
 #include "stdio_.h"
@@ -45,6 +47,7 @@
 pipe_fopen(gx_io_device * iodev, const char *fname, const char *access,
 	   FILE ** pfile, char *rfname, uint rnamelen)
 {
+#if 0
     errno = 0;
     /*
      * Some platforms allow opening a pipe with a '+' in the access
@@ -61,6 +64,9 @@
 	return_error(gs_fopen_errno_to_code(errno));
     if (rfname != NULL)
 	strcpy(rfname, fname);
+#endif
+
+    abort ();
     return 0;
 }
 
