--- fluidsynth-1.1.0/configure.ac~	2009-10-30 20:18:30.000000000 +0100
+++ fluidsynth-1.1.0/configure.ac	2009-11-10 09:17:46.944049001 +0100
@@ -71,7 +71,7 @@ mingw*)
   mingw32_support="yes"
   CFLAGS="$CFLAGS -mms-bitfields"
   AC_DEFINE(MINGW32, 1, [Define if using the MinGW32 environment])
-  LIBFLUID_LIBS="-ldsound -lwinmm -lws2_32"
+  LIBFLUID_LIBS="-static -ldsound -lwinmm -lws2_32"
   LIBFLUID_CPPFLAGS="-DFLUIDSYNTH_DLL_EXPORTS"
   LIBFLUID_LDFLAGS="-no-undefined"
   FLUID_CPPFLAGS="-DFLUIDSYNTH_NOT_A_DLL"
