--- ./fontforge/Makefile.in~	Thu Apr 13 19:39:21 2006
+++ ./fontforge/Makefile.in	Tue May  2 14:29:14 2006
@@ -61,9 +61,9 @@
 	$(LIBTOOL) --mode=link $(CC) -export-dynamic -o fontforge main.o $(fontforge_LIBOBJECTS) $(LIBS)
 
 sfddiff: sfddiff.pe
-	echo "#!$(bindir)/fontforge" >sfddiff
-	cat sfddiff.pe >>sfddiff
-	chmod +x sfddiff
+	echo "#!$(bindir)/fontforge" >$@
+	cat $^ >>$@
+	chmod +x $@
 
 sfddiff.old: $(DIFFOBJS)
 	$(LIBTOOL) --mode=link $(CC) -o sfddiff.old $(DIFFOBJS) $(DLIBS)
