diff -purN -x CYGWIN-PATCHES -x'*~' -x 'cyg*' -xjpeg -x'*.orig' ../../ghostscript-8.15/src/gp_unifn.c ./src/gp_unifn.c
--- ../../ghostscript-8.15/src/gp_unifn.c	2004-01-06 00:32:04.000000000 +0100
+++ ./src/gp_unifn.c	2004-12-07 07:38:59.000000000 +0100
@@ -29,8 +29,8 @@ const char gp_file_name_list_separator =
 const char gp_fmode_binary_suffix[] = "";
 
 /* Define the file modes for binary reading or writing. */
-const char gp_fmode_rb[] = "r";
-const char gp_fmode_wb[] = "w";
+const char gp_fmode_rb[] = "rb";
+const char gp_fmode_wb[] = "wb";
 
 /* -------------- Helpers for gp_file_name_combine_generic ------------- */
 
