--- libc/sysdeps/generic/wordexp.c.jj	2005-01-26 18:22:24.000000000 +0100
+++ libc/sysdeps/generic/wordexp.c	2005-03-15 23:12:58.569452725 +0100
@@ -809,7 +809,7 @@ parse_arith (char **word, size_t *word_l
 }
 
 /* Function called by child process in exec_comm() */
-static void
+static inline void
 internal_function __attribute__ ((always_inline))
 exec_comm_child (char *comm, int *fildes, int showerr, int noexec)
 {
