--- include/libc-symbols.h~	2003-01-04 00:21:03.000000000 +0100
+++ include/libc-symbols.h	2007-04-16 09:47:58.000000000 +0200
@@ -247,7 +247,7 @@
     __attribute__ ((unused, section (".gnu.warning." #symbol __sec_comment))) \
     = msg;
 #  define libc_freeres_ptr(decl) \
-  __make_section_unallocated ("__libc_freeres_ptrs, \"aw\", @nobits") \
+  __make_section_unallocated ("__libc_freeres_ptrs, \"aw\", %nobits") \
   decl __attribute__ ((section ("__libc_freeres_ptrs" __sec_comment)))
 #  define __libc_freeres_fn_section \
   __attribute__ ((section ("__libc_freeres_fn")))
