Index: gs/src/sfxstdio.c
===================================================================
--- gs/src/sfxstdio.c	(revision 7001)
+++ gs/src/sfxstdio.c	(working copy)
@@ -155,6 +155,7 @@
 	return 0;
     s_init_ids(s);
     s->is_temp = 0;		/* not a temp stream */
+    s->foreign = 0;
     /*
      * Disable the stream now (in case we can't open the file,
      * or a filter init procedure fails) so that `restore' won't
Index: gs/src/zcontrol.c
===================================================================
--- gs/src/zcontrol.c	(revision 7022)
+++ gs/src/zcontrol.c	(working copy)
@@ -1009,12 +1009,13 @@
 	es_ptr ep = rsenum.ptr + used - 1;
 	uint count = used;
 
-	for (; count; count--, ep--)
-	    if (r_is_estack_mark(ep) &&
-		estack_mark_index(ep) == es_stopped &&
-		(ep[2].value.intval & mask) != 0
-		)
-		return scanned + (used - count + 1);
+	for (; count; count--, ep--) {
+	    if (r_is_estack_mark(ep)) {
+		if (estack_mark_index(ep) == es_stopped &&
+		  (ep[2].value.intval & mask) != 0)
+		    return scanned + (used - count + 1);
+	    }
+	}	
 	scanned += used;
     } while (ref_stack_enum_next(&rsenum));
     return 0;
