--- gzip-1.3.12/gzip.c	2007-03-20 05:09:51.000000000 +0000
+++ gzip-1.3.12/gzip.c	2009-08-21 00:13:55.000000000 +0100
@@ -1637,7 +1637,7 @@
 	}
       }
 
-    if (futimens (ofd, ofname, timespec) != 0)
+    if (gz_futimens (ofd, ofname, timespec) != 0)
       {
 	int e = errno;
 	WARN ((stderr, "%s: ", program_name));
--- gzip-1.3.12/lib/utimens.c	2007-01-18 08:33:34.000000000 +0000
+++ gzip-1.3.12/lib/utimens.c	2009-08-21 00:13:55.000000000 +0100
@@ -75,7 +75,7 @@
    Return 0 on success, -1 (setting errno) on failure.  */
 
 int
-futimens (int fd ATTRIBUTE_UNUSED,
+gz_futimens (int fd ATTRIBUTE_UNUSED,
 	  char const *file, struct timespec const timespec[2])
 {
   /* Some Linux-based NFS clients are buggy, and mishandle time stamps
@@ -185,5 +185,5 @@
 int
 utimens (char const *file, struct timespec const timespec[2])
 {
-  return futimens (-1, file, timespec);
+  return gz_futimens (-1, file, timespec);
 }
--- gzip-1.3.12/lib/utimens.h	2007-02-23 18:25:21.000000000 +0000
+++ gzip-1.3.12/lib/utimens.h	2009-08-21 00:13:55.000000000 +0100
@@ -1,3 +1,3 @@
 #include <time.h>
-int futimens (int, char const *, struct timespec const [2]);
+int gz_futimens (int, char const *, struct timespec const [2]);
 int utimens (char const *, struct timespec const [2]);
