--- inkscape-trunk-20605/src/ege-adjustment-action.cpp~	2009-02-09 11:54:42.000000000 +0100
+++ inkscape-trunk-20605/src/ege-adjustment-action.cpp	2009-02-11 12:25:25.000000000 +0100
@@ -62,6 +62,7 @@
 
 #include "ege-adjustment-action.h"
 
+#include "config.h"
 
 
 static void ege_adjustment_action_class_init( EgeAdjustmentActionClass* klass );
--- inkscape-trunk-20605/src/display/pixblock-transform.cpp~	2009-02-09 11:54:42.000000000 +0100
+++ inkscape-trunk-20605/src/display/pixblock-transform.cpp	2009-02-11 12:25:25.000000000 +0100
@@ -25,6 +25,8 @@
 #include "libnr/nr-pixblock.h"
 #include <2geom/matrix.h>
 
+#include "config.h"
+
 namespace NR {
 
 struct RGBA {
--- inkscape-trunk-20605/src/2geom/convex-cover.cpp~	2009-02-09 11:54:42.000000000 +0100
+++ inkscape-trunk-20605/src/2geom/convex-cover.cpp	2009-02-11 12:25:25.000000000 +0100
@@ -34,6 +34,8 @@
 #include <algorithm>
 #include <map>
 
+#include "config.h"
+
 /** Todo:
     + modify graham scan to work top to bottom, rather than around angles
     + intersection
