--- lpsolve-5.5.0.13.source/lp_types.h.orig	2008-10-29 21:50:54.000000000 +0100
+++ lpsolve-5.5.0.13.source/lp_types.h	2008-10-29 21:51:28.000000000 +0100
@@ -182,7 +182,7 @@
 #if !defined INLINE
   #if defined __cplusplus
     #define INLINE inline
-  #elif (defined _WIN32) || (defined WIN32) || (defined _WIN64) || (defined WIN64)
+  #elif ((defined _WIN32) || (defined WIN32) || (defined _WIN64) || (defined WIN64)) && ! defined (__MINGW32__)
     #define INLINE __inline
   #else
     #define INLINE static
