diff -ur nsis-2.19-src/Source/script.cpp nsis-2.19-src.patched/Source/script.cpp
--- nsis-2.19-src/Source/script.cpp	2006-08-20 02:55:11.000000000 +0200
+++ nsis-2.19-src.patched/Source/script.cpp	2006-08-20 02:55:32.000000000 +0200
@@ -4744,18 +4744,9 @@
         struct stat st;
         if (!stat(line.gettoken_str(1), &st))
         {
-          union
-          {
-            struct
-            {
-              long l;
-              long h;
-            } words;
-            long long ll;
-          };
-          ll = (st.st_mtime * 10000000LL) + 116444736000000000LL;
-          high = words.h;
-          low = words.l;
+          unsigned long long ll = (st.st_mtime * 10000000LL) + 116444736000000000LL;
+          high = (DWORD) (ll >> 32);
+          low = (DWORD) ll;
         }
         else
         {
@@ -6071,22 +6062,13 @@
       struct stat st;
       if (!fstat(fd, &st))
       {
-        union
-        {
-          struct
-          {
-            long l;
-            long h;
-          } words;
-          long long ll;
-        };
-        ll = (st.st_mtime * 10000000LL) + 116444736000000000LL;
+        unsigned long long ll = (st.st_mtime * 10000000LL) + 116444736000000000LL;
 
         // FAT write time has a resolution of 2 seconds
         ll -= ll % 20000000;
 
-        ent.offsets[3] = words.l;
-        ent.offsets[4] = words.h;
+        ent.offsets[3] = (int) ll;
+        ent.offsets[4] = (int) (ll >> 32);
       }
 #endif
       else
Only in nsis-2.19-src.patched/Source: script.cpp~
