--- SConstruct~	2006-02-24 20:19:49.000000000 +0100
+++ SConstruct	2006-03-23 14:24:52.000000000 +0100
@@ -151,7 +151,7 @@
 f = open(File('#Source/exehead/sconf.h').abspath, 'w')
 for i in defenv['NSIS_CPPDEFINES']:
 	if type(i) is not str:
-		f.write('#define %s %s\n' % (i[0], i[1]))
+		f.write('#define %s %s\n' % (i[0], defenv.subst (i[1])))
 	else:
 		f.write('#define %s\n' % (i))
 f.close()
