--- config_office/configure.in.orig	2008-10-19 22:03:22.000000000 +0200
+++ config_office/configure.in	2008-10-19 22:17:08.000000000 +0200
@@ -1059,7 +1079,7 @@
 fi
 AC_SUBST(ENABLE_CRASHDUMP)
 
-if test "$_os" = "WINNT"; then
+if test "$_os" = "WINNT" -o "$t_os" = "MinGW"; then
    BUILD_TYPE="$BUILD_TYPE TWAIN"
 fi
 
@@ -4265,7 +4262,7 @@ elif test "$_os" = "OS2" ; then
    echo "Do Nothing for _os = OS2. Don't check for X11."
    dnl OS/2 uses native graphics. Don't check for X11.
    :
-elif test "$_os" != "WINNT" ; then
+elif test "$_os" != "WINNT" -a "$t_os" != "MinGW" ; then
    AC_PATH_X
    AC_PATH_XTRA
    CPPFLAGS="$CPPFLAGS $X_CFLAGS"
--- config_office/set_soenv.in.orig	2008-10-20 19:52:27.000000000 +0200
+++ config_office/set_soenv.in	2008-10-20 20:22:58.000000000 +0200
@@ -593,6 +593,22 @@ elsif ( $platform =~ m/linux/ ) 
       $JREEXTRALIBDIR = '$JAVA_HOME'.$ds."lib".$ds.$ARCH.$ds."xawt";
    }
 
+   if ( "@TARGET_OS@" == "MinGW" )
+   {
+       print "Setting build=Linux -> target,host=MinGW specific values... ";
+       $COM            = "GCC";
+       $CVER           = "C341";
+       $OUTPATH        = "wntgcci";
+       $CPUNAME        = "INTEL";
+       $CPU            = "I";
+       $GUI            = "WNT";
+       $GUIBASE        = "WIN";
+       $GVER           = "NT351";
+       $OS             = "WNT";
+       $DYNAMIC_CRT    = "TRUE";
+       $SET_EXCEPTIONS = "TRUE";
+       $use_shl_versions = "TRUE";
+   }
 
    $INPATH         = $OUTPATH.$PROEXT;
 }
