--- setup_native/source/win32/customactions/tools/seterror.cxx.pristine	2008-10-28 20:55:13.000000000 +0100
+++ setup_native/source/win32/customactions/tools/seterror.cxx	2008-10-28 21:12:01.000000000 +0100
@@ -41,12 +41,12 @@
 #include <string.h>
 #include <malloc.h>
 #include <stdio.h>
-#include "strsafe.h"
 
 #include <seterror.hxx>
 
 //----------------------------------------------------------
 #ifdef DEBUG
+#include "strsafe.h"
 inline void OutputDebugStringFormat( LPCTSTR pFormat, ... )
 {
 	TCHAR    buffer[1024];
--- setup_native/source/win32/stwrapper/resource.rc.pristine	2008-10-28 21:24:11.000000000 +0100
+++ setup_native/source/win32/stwrapper/resource.rc	2008-10-28 21:24:37.000000000 +0100
@@ -113,7 +113,11 @@ VS_VERSION_INFO versioninfo
 VS_VERSION_INFO rcdata
 {
 	0xF0, "sw", 0x0F, VER_YEAR, VER_MONTH, VER_DAY,
+#ifndef CROSS_COMPILING
 	VERSION, SUBVERSION, VERVARIANT, VER_COUNT
+#else
+	VERSION, SUBVERSION
+#endif
 };
 
 
