--- connectivity/source/inc/odbc/OFunctiondefs.hxx.~1.17.~	2008-04-10 13:43:33.000000000 +0200
+++ connectivity/source/inc/odbc/OFunctiondefs.hxx	2008-10-31 11:39:30.000000000 +0100
@@ -32,7 +32,14 @@
 #ifndef _CONNECTIVITY_OFUNCTIONDEFS_HXX_
 #define _CONNECTIVITY_OFUNCTIONDEFS_HXX_
 
-#if defined(WIN) || defined(WNT)
+#if defined(__MINGW32__)
+
+#include <windows.h>
+#define SDB_ODBC_CHAR SQLCHAR
+#include <odbcinst.h>
+#include <sqlext.h>
+
+#elif defined(WIN) || defined(WNT)
 
 #ifdef _MSC_VER
 #pragma warning(push)
