--- tools/source/fsys/wntmsc.hxx.pristine	2008-10-28 19:13:52.000000000 +0100
+++ tools/source/fsys/wntmsc.hxx	2008-10-28 19:13:59.000000000 +0100
@@ -36,8 +36,8 @@
 #ifndef ICC
 #include <io.h>
 #endif
-#include <sys\types.h>
-#include <sys\stat.h>
+#include <sys/types.h>
+#include <sys/stat.h>
 #include <direct.h>
 
 #include <tools/svwin.h>
--- tools/inc/tools/solar.h.pristine	2008-10-29 23:44:36.000000000 +0100
+++ tools/inc/tools/solar.h	2008-10-30 19:56:43.000000000 +0100
@@ -57,6 +57,15 @@
 /*   Previous typedefs from before sal convergence are in comments */
 /*   to the right of the new types.                                */
 
+#ifdef __MINGW32__
+#undef BOOL
+#define BOOL OOO_BOOL
+#undef INT32
+#define INT32 OOO_INT32
+#undef UINT32
+#define UINT32 OOO_UINT32
+#endif /* __MINGW32__ */
+
 typedef sal_Bool		BOOL;	/* typedef unsigned char   BOOL; */
 typedef sal_uInt8		BYTE;	/* typedef unsigned char   BYTE; */
 typedef sal_uInt16		USHORT;	/* typedef unsigned short  USHORT; */
