--- tools/source/fsys/wntmsc.hxx.pristine	2008-10-28 19:13:52.000000000 +0100
+++ tools/source/fsys/wntmsc.hxx	2008-10-28 19:13:59.000000000 +0100
@@ -36,8 +36,8 @@
 #ifndef ICC
 #include <io.h>
 #endif
-#include <sys\types.h>
-#include <sys\stat.h>
+#include <sys/types.h>
+#include <sys/stat.h>
 #include <direct.h>
 
 #include <tools/svwin.h>
--- tools/inc/tools/solar.h.pristine	2008-10-29 23:44:36.000000000 +0100
+++ tools/inc/tools/solar.h	2008-10-30 20:21:41.000000000 +0100
@@ -57,7 +57,10 @@
 /*   Previous typedefs from before sal convergence are in comments */
 /*   to the right of the new types.                                */
 
+
+#ifndef BOOL
 typedef sal_Bool		BOOL;	/* typedef unsigned char   BOOL; */
+#endif
 typedef sal_uInt8		BYTE;	/* typedef unsigned char   BYTE; */
 typedef sal_uInt16		USHORT;	/* typedef unsigned short  USHORT; */
 typedef sal_uIntPtr		ULONG;	/* typedef unsigned long   ULONG; */
@@ -81,8 +84,12 @@ typedef sal_uInt8			UINT8;	/* typedef un
 /* types with exact defined size (not just the minimum size) */
 typedef sal_Int64			INT64;	/* typedef long            INT64;  previously "void" on Windows */
 typedef sal_uInt64			UINT64;	/* typedef unsigned long   UINT64; previously "void" on Windows */
+#ifndef INT32
 typedef sal_Int32			INT32;	/* typedef int             INT32;  previously "long" on Windows */
+#endif
+#ifndef UINT32
 typedef sal_uInt32			UINT32;	/* typedef unsigned int    UINT32; previously "unsigned long" on Windows */
+#endif
 #endif /* _SOLAR_NOUINT */
 #endif
 
