--- Makefile.orig	2007-04-13 08:40:09.000000000 +0200
+++ Makefile	2007-04-13 08:45:45.000000000 +0200
@@ -57,3 +57,14 @@
 		cp /tmp/id $$f; \
 	done
 
+prefix = /usr/local
+install:
+	mkdir -p $(DESTDIR)$(prefix)/lib
+	cp -L $$(find . -name '*.a') $(DESTDIR)$(prefix)/lib
+	mkdir -p $(DESTDIR)$(prefix)/include
+	cp -RL $$(find  . -name include) $(DESTDIR)$(prefix)
+	cd $(DESTDIR)$(prefix)/lib && for i in $$(find . -name 'libpj*a'); do\
+		ln -s $$i $$(echo $$i | sed -e "s/-$(TARGET_NAME)//");\
+	done
+	mkdir -p $(DESTDIR)$(prefix)/lib/pkgconfig
+	sed -e "s!@PREFIX@!$(DESTDIR)$(prefix)!" libpj.pc.in > $(DESTDIR)/$(prefix)/lib/pkgconfig/libpj.pc
--- libpj.pc.in.orig	1970-01-01 01:00:00.000000000 +0100
+++ libpj.pc.in	2007-03-16 10:49:27.000000000 +0100
@@ -0,0 +1,12 @@
+# Package Information for pkg-config
+
+prefix=@PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+
+Name: libpj
+Description: Sip library
+Version: 5.10.1
+Libs: -L${libdir} -lpjsua -lpjsip -lpjmedia -lpjsip-ua -lpjsip-simple -lpjsip-ua -lpjmedia-codec -lpjlib-util -lpj
+Cflags: -I${includedir}
