From 7e564cfda79a5777ce29b6f5e39ef19a900c94da Mon Sep 17 00:00:00 2001
From: Han-Wen Nienhuys <hanwen@lilypond.org>
Date: Sat, 14 Feb 2009 15:36:52 -0200
Subject: [PATCH] Disable readline setup.

---
 setup.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/setup.py b/setup.py
index bb7b26e..738b9a9 100644
--- a/setup.py
+++ b/setup.py
@@ -439,7 +439,7 @@ class PyBuildExt(build_ext):
             exts.append( Extension('rgbimg', ['rgbimgmodule.c']) )
 
         # readline
-        if self.compiler.find_library_file(lib_dirs, 'readline'):
+        if False and self.compiler.find_library_file(lib_dirs, 'readline'):
             readline_libs = ['readline']
             if self.compiler.find_library_file(lib_dirs,
                                                  'ncursesw'):
-- 
1.6.0.6

