--- setup.py~	Thu Mar  9 15:36:59 2006
+++ setup.py	Thu Mar  9 16:01:34 2006
@@ -174,6 +174,13 @@
         # compilers
         if compiler is not None:
             (ccshared,opt,base) = sysconfig.get_config_vars('CCSHARED','OPT','BASECFLAGS')
+
+            ##
+            ## Has anyone with python development ever heard of cross-compiling?! Bunch of morons!
+            import re
+            for forbidden in ['-Wno-long-double', '-no-cpp-precomp', '-mno-fused-madd']:
+                base = re.sub (forbidden, ' ', base)
+
             args['compiler_so'] = compiler + ' ' + opt + ' ' + ccshared + ' ' + base
         self.compiler.set_executables(**args)
 
