--- python-2.4.5/Modules/Setup.dist.orig	2008-11-11 11:52:04.000000000 +0100
+++ python-2.4.5/Modules/Setup.dist	2008-11-11 11:52:20.000000000 +0100
@@ -111,8 +111,8 @@ PYTHONPATH=$(COREPYTHONPATH)
 
 @POSIX@ posixmodule.c		# posix (UNIX) system calls
 errno errnomodule.c		# posix (UNIX) errno values
-pwd pwdmodule.c			# this is needed to find out the user's home dir
-				# if $HOME is not set
+#pwd pwdmodule.c			# this is needed to find out the user's home dir
+#				# if $HOME is not set
 _sre _sre.c			# Fredrik Lundh's new regular expressions
 _codecs _codecsmodule.c		# access to the builtin codecs and codec registry
 
